src/storage/exec/RelNode.h

```diff
@@ -68,7 +68,7 @@ class RelNode {
 
   void addDependency(RelNode<T>* dep) {
     dependencies_.emplace_back(dep);
-    dep->hasDependents_ = true;
+    dep->isRoot_ = false;
```

Comment:
`hasDependents_` indicates that other nodes depend on this node.
This node must be executed first, and then the other can be executed.
I feel `hasDependents_`  is  clearer?

Reply:
First, this function `addDependency ` will add all `dep` to the member vector `dependencies_ `. From this alone, it is natural to understand that `this` node depends on `dep`. From the execution flow, it can also be verified. `GetNeighbor` depends on `hashJoin`, depends on `Edge`.... But it conflicts with `dep->hasDependents_ = true;` This line is saying `dep` is depending on something, rather than `this` depends on `dep`. However, let's take an example, in `GetNeighborsProcessor`, the `Edge` node does not depend on anything in the execution flow. It is the leaf node. But with the current logic, `Edge` node will be set `hasDependents_`.

Reply:
you're right.
The true semantics of `hasDependents_` is to be dependent.